### PR TITLE
change capabilities logic for sudo prefixes

### PR DESCRIPTION
**Pull request by madalynrose**
specifically for sys/leases/revoke with update capability

### CODE REVIEW

File: ui/tests/unit/models/capabilities-test.js [hunk 1]
```diff
@@ -81,4 +81,18 @@ module('Unit | Model | capabilities', function(hooks) {
...
+
+  test('it does not require sudo on sys/leases/revoke if update capability is present', function(assert) {
+    let model = run(() =>
+      this.owner.lookup('service:store').createRecord('capabilities', {
+        path: 'sys/leases/revoke',
+        capabilities: ['update', 'read'],
+      })
+    );
+    assert.ok(model.get('canRead'));
+    assert.notOk(model.get('canCreate'), 'sudo requires the capability to be set as well');
```
Comment (meirish) [inline ui/tests/unit/models/capabilities-test.js]: I think this doesn't have to do with sudo, just that `create` wasn't in the capabilities list.

Comment (madalynrose) [reply ui/tests/unit/models/capabilities-test.js]: it's in every test so I just left it

File: ui/tests/unit/models/capabilities-test.js [hunk 2]
```diff
@@ -81,4 +81,18 @@ module('Unit | Model | capabilities', function(hooks) {
...
+    assert.ok(model.get('canUpdate'), 'should not require sudo if it has update');
+    assert.notOk(model.get('canDelete'));
+    assert.notOk(model.get('canList'));
+  });
```
Comment (meirish) [inline ui/tests/unit/models/capabilities-test.js]: Can we add another test for the paths in the SUDO_PREFIX arg too to make sure they _are_ requiring sudo?

So for `path: 'sys/leases/revoke-prefix/aws'` with capabilities "update", `canUpdate` should be false, but with "update", "sudo", `canUpdate` should be true.

Comment (madalynrose) [reply ui/tests/unit/models/capabilities-test.js]: I added a couple. Do they look right?

Comment (meirish) [reply ui/tests/unit/models/capabilities-test.js]: Yassss - perfect!

**Pull request merged**